### ISSUE TITLE
[Feature request] A method to highlight clues needing review

### ISSUE THREAD

**Issue by Antagony1060** [viresh-ratnakar/exet#10]
I created another puzzle) with Exet and at first I tried just using it on its own, but one thing I really missed from my old way of working was being able to highlight clues — written in OneNote — so I could tell at a glance which ones I was still not entirely happy with. I used to use text background highlighting, but it was a bit of a pain having to select the text first, so more recently I just inserted a checkbox next to each clue — which can be done easily in OneNote with Ctrl+1 — and only ticked it when satisfied with the clue.

I think it could be even more intuitive in Exet, as you could also shade the clues — or their fonts — in the grid. It would also probably make sense to mark all clues that way in new grids.

What do you think?

**Comment by viresh-ratnakar** [comment]
Have done this now, with v0.32, thanks for the suggestion—it was kinda fun to work through this.

I used to leave the "TO DO" (that was a default prefix for clues) in place within the clue until a clue was ready, as a distinctive marker. But I think having an explicit, actionable tracking mechanism makes sense. However, I do like the simplicity of a prefix in the clue (which also works when reading puzzles from Exolve files without having to create a separate mechanism for keeping track of which clues still need work). But "TO DO" is perhaps a bit eclectic.

So, I've implemented keeping track of clues in "draft" mode by now making new clues start out with a "[DRAFT]" prefix (clearer than "TO DO," hopefully). I've also added code to be aware of this prefix and translate it into distinctive rendering (including a gray clue number in the list of clues). And there is a button in the clue editing panel now, that lets you toggle the draft mode. Further, the "Save" menu shows you a warning message if there still are clues in draft mode.

I went through several variants when working on this, including using check marks for marking clues no longer in draft mode. But I converged on what I've checked in as I really want the clues not in draft mode to just look "normal" rather than have any distinctive markings.

But PTAL, am happy to listen to ideas for rendering differently.
